book/src/pong_tutorial/pong_tutorial_02.md

```diff
@@ -0,0 +1,367 @@
+[The Specs Book][sb].
+
+## A quick refactor
+
+Let's create a new file called `pong.rs` to hold our core game logic. We can 
+move the `Pong` struct over here, and the `impl State for Pong` block as well. 
+Then, in `main.rs` declare a module:
+
+```rust,ignore
+mod pong; 
+```
+
+And in the `run()` function add:
+
+```rust,ignore
+use pong::Pong;
+```
+
+Now you can just delete various `main.rs` use statements until the Rust compiler 
+stops complaining about unused imports. In `pong.rs` we'll need these use 
+statements to make it through this chapter:
+
+```rust,ignore
+use amethyst::prelude::*;
+use amethyst::ecs::{Component, DenseVecStorage};
+use amethyst::assets::Loader;
+use amethyst::core::cgmath::Vector3;
+use amethyst::core::transform::{LocalTransform, Transform};
+use amethyst::renderer::{Camera, Material, MaterialDefaults, PosTex, MeshHandle, 
+                         Event,KeyboardInput, VirtualKeyCode, WindowEvent};
+```
+
+## Our first Component
+
+In pong.rs let's create our first `Component`, a definition of a paddle.
+
+
+```rust,ignore
+#[derive(PartialEq, Eq)]
+enum Side {
+    Left,
+    Right,
+}
+
+struct Paddle {
+    pub side: Side,
+    pub width: f32,
+    pub height: f32,
+}
+
+impl Paddle {
+    fn new(side: Side) -> Paddle {
+        Paddle {
+            side: side,
+            width: 1.0,
+            height: 1.0,
+        }
+    }
+}
+```
+
+"But that just looks like a regular struct!" you might say. And you're right, 
+here's the special sauce:
+
+```rust,ignore
+impl Component for Paddle {
+    type Storage = DenseVecStorage<Self>;
+}
+```
+
+By implementing `Component` for our `Paddle` struct, and defining the way we'd 
+like that `Component` data stored, we can now add the `Paddle` component to 
+entities in our game. For more on storage types, check out the 
+[Specs documentation][sb-storage].
+
+## Where is the world?
+
+Now it's time to add a new method to our State implementation: `on_start`. 
+Inside this function, we'll `register` our `Paddle` component on the mutable 
+`World` object we're passed by Amethyst's state machine when the game starts up.
+
+```rust,ignore
+fn on_start(&mut self, world: &mut World) {
+  world.register::<Paddle>();
+}
+```
+
+This `World` gets passed around everywhere. It carries with it all the 
+components and resources in our game. Not only the components and resources we 
+create, but the ones the Amethyst engine itself relies on. For instance, in our 
+`main.rs` we added a `RenderBundle::new()` to our game before calling `run()`. 
+That added default rendering components like `Camera`, `Material`, and `Mesh` to 
+the `World`, some of which we'll be using soon.
+
+## Initialise some entities
+
+Now that we have a Paddle component, let's define some paddle entities that 
+include that component and add them to our `World`.
+
+First let's look at our math imports:
+
+```rust,ignore
+use amethyst::core::cgmath::{Vector3};
```

Comment:
Proper rust style here would be to omit the `{` and `}` until we have more than element to import here.